main.go

```diff
@@ -145,3 +185,76 @@ func loadConfig(cfgJSON *extapi.JSON) (customDNSProviderConfig, error) {
 
 	return cfg, nil
 }
+
+func (c *ns1DNSProviderSolver) setNS1Client(ch *v1alpha1.ChallengeRequest, cfg ns1DNSProviderConfig) error {
+	ref := cfg.APIKeySecretRef
+	if ref.Key == "" {
+		return fmt.Errorf(
+			"no key '%s' in secret '%s/%s'",
+			ref.Key,
+			ch.ResourceNamespace,
+			ref.Name,
+		)
+	}
+	if ref.Name == "" {
+		return fmt.Errorf(
+			"no Key '%s' in secret '%s/%s'",
+			ref.Key,
+			ch.ResourceNamespace,
+			ref.Name,
+		)
+	}
+
+	secret, err := c.k8sClient.CoreV1().Secrets(ch.ResourceNamespace).Get(
+		ref.Name, metav1.GetOptions{},
+	)
+	if err != nil {
+		return err
+	}
+	apiKeyBytes, ok := secret.Data[ref.Key]
+	if !ok {
+		return fmt.Errorf(
+			"no key '%s' in secret '%s/%s'",
+			ref.Key,
+			ch.ResourceNamespace,
+			ref.Name,
+		)
+	}
+	apiKey := string(apiKeyBytes)
+
+	httpClient := &http.Client{}
+	if cfg.IgnoreSSL == true {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		httpClient.Transport = tr
+	}
+	c.ns1Client = ns1API.NewClient(
+		httpClient,
+		ns1API.SetAPIKey(apiKey),
+		ns1API.SetEndpoint(cfg.Endpoint),
```

Comment:
Can we give `cfg.endpoint` a default value if it's not provided (or only set it if it _is_ provided)?  Given the current install base of NS1, most folks will probably be using this with managed, right?

Reply:
agree these (ignorreSSL also) could/should be optional with default. thought about that but things were taking a while and i thought i'd try to get this initial guy merged first.